stdlib/public/core/StringBridge.swift

```diff
@@ -590,7 +619,8 @@ extension String {
 
     _internalInvariant(_guts._object.hasObjCBridgeableObject,
       "Unknown non-bridgeable object case")
-    return _guts._object.objCBridgeableObject
+    let result = _guts._object.objCBridgeableObject
+    return formConstantTaggedCocoaString(untaggedCocoa: result) ?? result
```

Comment:
Here we'd be checking a spare perf flag on the second word instead.

Reply:
Yeah. I think that's ultimately where we want to end up.